androidApp/src/main/java/com/example/tasky/android/agenda/viewmodel/AgendaDetailsViewModel.kt

```diff
@@ -276,7 +328,15 @@ class AgendaDetailsViewModel(
         }
     }
 
-    private fun saveUpdate() {
+    private fun saveAgenda() {
+        if (routeArguments.agendaId.isEmpty()) {
+            createRemoteAgenda()
+        } else {
+            updateRemoteAgenda()
+        }
```

Comment:
I found that creating a new agenda item when the user taps on the “+” (and selects a task/event/reminder) with default values (which you already have) and saving it to your local db simplifies the detail screen’s logic. You can create the model and save it to the local DB. Once done, you can pass the ID as a navigation argument to the Detail screen and load it again from your DB. 

For me, this is a simpler approach than trying to manage an in-memory model. I just reflect exactly what is in the local DB. Any changes the user makes get saved to the local DB, and then I get a new flow emission because the model in the database changes, and my UI is automatically updated.